### DISCUSSION TITLE
Certificate Object

### DISCUSSION THREAD

**Discussion by Aniak5** [ocsf/ocsf-schema#63] [General]
Propose to move the `certificate` object from the AWS extensions to the main `objects` directory and replace the `pem` attribute within it to either `fingerprint` or `cert_uid` (leaning towards `fingerprint`). Within the `tls.json` object remove `    "cert_serial": {
      "requirement": "optional"
    },
    "cert_uid": {
      "requirement": "recommended"
    },` and add the new `certificate` object. Replace `domain` in the TLS object and replace it with a new attribute `sni`.

**Comment by floydtree** [comment]
All the changes sound good to me. I agree, `fingerprint` would be better.

**Reply by Aniak5** [reply]
I see that the `access_logs` event class within the `Access Application` category uses the `certificate` object. Does anyone have any thoughts on transferring or using pieces of this event within main?

**Reply by floydtree** [reply]
We initially created `access_logs` when we were looking at a few access logs types that we have in AWS, it's fairly nascent as of now, but we can definitely port it over to the core & update it if you see value in that class.
